### PR TITLE
Change IModelCombiner to not be generic, and add unit tests

### LINKED ISSUE #1304: Need a non-generic version of the IModelCombiner interface
In order to be able to define arguments that are factories of "IModelCombiner" where any kind of IModelCombiner can be specified, we need a non-generic version of this interface.

### PR DESCRIPTION
**Pull request by yaeldekel**
Fixes #1304.

### CODE REVIEW

File: test/Microsoft.ML.Predictor.Tests/TestPredictors.cs [hunk 1]
```diff
@@ -649,62 +651,288 @@ private void CombineAndTestTreeEnsembles(IDataView idv, IPredictorModel[] fastTr
...
+                    LabelColumn = DefaultColumnNames.Label,
+                    OptTol = 10e-3F,
+                    TrainingData = dataView,
+                    NormalizeFeatures = NormalizeOption.No
+                }).PredictorModel
+            };
+            CombineAndTestEnsembles(dataView, "pe", "oc=average", PredictionKind.BinaryClassification, predictors);
+        }
+
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // x86 fails with "Unknown command: 'train'; Format error at (83,3)-(83,4011): Illegal quoting"
```
Comment (Ivanidzo4ka) [inline test/Microsoft.ML.Predictor.Tests/TestPredictors.cs]: >// x86 fails with "Unknown command: 'train'; Format error at (83,3)-(83,4011): Illegal quoting" [](start = 83, length = 95)

Can we have issue about this? it's weird exception, I don't get how change of platform can lead to change of reading data.

File: test/Microsoft.ML.Predictor.Tests/TestPredictors.cs [hunk 2]
```diff
@@ -649,62 +651,288 @@ private void CombineAndTestTreeEnsembles(IDataView idv, IPredictorModel[] fastTr
...
+        public void TestEnsembleCombiner()
+        {
+            var dataPath = GetDataPath("breast-cancer.txt");
+            var inputFile = new SimpleFileHandle(Env, dataPath, false, false);
+#pragma warning disable 0618
+            var dataView = ImportTextData.ImportText(Env, new ImportTextData.Input { InputFile = inputFile }).Data;
+#pragma warning restore 0618
```
Comment (Ivanidzo4ka) [inline test/Microsoft.ML.Predictor.Tests/TestPredictors.cs]: Can we not add obsolete code? 
```
            var reader = TextLoader.CreateReader(Env, ctx => (
                Features: ctx.LoadFloat(1, 8),
                Label: ctx.LoadFloat(0)
            ));
            var data = reader.Read(dataPath).AsDynamic;

```

File: src/Microsoft.ML.Ensemble/Trainer/Binary/EnsembleTrainer.cs [hunk 3]
```diff
@@ -70,6 +73,12 @@ public EnsembleTrainer(IHostEnvironment env, Arguments args)
             Combiner = args.OutputCombiner.CreateComponent(Host);
         }
 
+        public EnsembleTrainer(IHostEnvironment env, Arguments args, PredictionKind predictionKind)
```
Comment (sfilipi) [inline src/Microsoft.ML.Ensemble/Trainer/Binary/EnsembleTrainer.cs]: >  public EnsembleTrainer(IHostEnvironment env, Arguments args, PredictionKind predictionKind) [](start = 6, length = 93)

in the spirit of reducing public ctor, would it make sense to add the PredictionKind as an optional param, or that would be confusing?

File: test/Microsoft.ML.Predictor.Tests/TestPredictors.cs [hunk 4]
```diff
@@ -649,62 +651,288 @@ private void CombineAndTestTreeEnsembles(IDataView idv, IPredictorModel[] fastTr
...
+            }
+            finally
+            {
+                for (int i = 0; i < predCount; i++)
+                    cursors[i].Dispose();
+            }
+        }
+
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // x86 fails with "Unknown command: 'train'; Format error at (83,3)-(83,4011): Illegal quoting"
+        public void TestEnsembleCombiner()
```
Comment (sfilipi) [inline test/Microsoft.ML.Predictor.Tests/TestPredictors.cs]: maybe log an issue about this?

File: src/Microsoft.ML.Ensemble/Trainer/Binary/EnsembleTrainer.cs (hunk 3 above)
Comment (yaeldekel) [reply src/Microsoft.ML.Ensemble/Trainer/Binary/EnsembleTrainer.cs]: I made it private :-).

---
In reply to: [227114506](https://github.com/dotnet/machinelearning/pull/1305#discussion_r227114506) [](ancestors = 227114506)

File: test/Microsoft.ML.Predictor.Tests/TestPredictors.cs (hunk 1 above)
Comment (yaeldekel) [reply test/Microsoft.ML.Predictor.Tests/TestPredictors.cs]: I copied this comment from the other tests in this file, I'm not sure it actually fails on x86.
@tannergooding , is there an issue for this?

Comment (tannergooding) [reply test/Microsoft.ML.Predictor.Tests/TestPredictors.cs]: https://github.com/dotnet/machinelearning/issues/1216

**Pull request merged**